packages/main/src/plugin/container-registry.ts

```diff
@@ -174,6 +176,27 @@ export class ContainerProviderRegistry {
     };
     let previousStatus = containerProviderConnection.status();
 
+    providerRegistry.onBeforeDidUpdateContainerConnection(event => {
+      if (event.providerId === provider.id && event.connection.name === containerProviderConnection.name) {
+        const newStatus = event.status;
+        if (newStatus === 'stopped') {
+          internalProvider.api = undefined;
+          internalProvider.libpodApi = undefined;
+          this.apiSender.send('provider-change', {});
+        }
+        if (newStatus === 'started') {
+          internalProvider.api = new Dockerode({ socketPath: containerProviderConnection.endpoint.socketPath });
+          if (containerProviderConnection.type === 'podman') {
+            internalProvider.libpodApi = internalProvider.api as unknown as LibPod;
+          }
+          this.handleEvents(internalProvider.api);
+          this.internalProviders.set(id, internalProvider);
+          this.apiSender.send('provider-change', {});
```

Comment:
with https://github.com/containers/podman-desktop/pull/2938/files you could just call setupConnectionAPI for 90% of these lines